include/seastar/coroutine/generator.hh

```diff
@@ -174,18 +176,11 @@ public:
     template<std::convertible_to<T> U>
     yield_awaiter<T, Container> yield_value(U&& value) noexcept {
         bool ready = _generator->put_next_value(std::forward<U>(value));
-
         if (_wait_for_next_value) {
             _wait_for_next_value->set_value();
             _wait_for_next_value = {};
-            return {make_ready_future()};
-        }
-        if (ready) {
-            return {make_ready_future()};
-        } else {
-            assert(!_wait_for_free_space);
-            return {_wait_for_free_space.emplace().get_future()};
         }
+        return {ready};
```

Comment:
Might as well delete yield_awaiter and return ready directly, no? The new yield_awaiter does nothing.

Reply:
again, i don't want to return to the caller coroutine. we can define `yield_awaiter` right in `generator_buffered_promise` if you prefer this way.